rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,153 @@
+package com.netflix.spinnaker.rosco.manifests.kustomize;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.rosco.jobs.BakeRecipe;
+import com.netflix.spinnaker.rosco.manifests.TemplateUtils;
+import com.netflix.spinnaker.rosco.manifests.kustomize.mapping.Kustomization;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+@Slf4j
+public class KustomizeTemplateUtils extends TemplateUtils {
+
+  private HashSet<String> filesToDownload = new HashSet<String>();
+  private static final String KUSTOMIZATION_FILE = "kustomization";
+
+  @Autowired
+  private KustomizationFileReader kustomizationFileReader;
+
+
+  public BakeRecipe buildBakeRecipe(BakeManifestEnvironment env, KustomizeBakeManifestRequest request) {
+    BakeRecipe result = new BakeRecipe();
+    result.setName(request.getOutputName());
+    filesToDownload = new HashSet<>();
+
+    List<Path> valuePaths = new ArrayList<>();
+    List<Artifact> inputArtifacts = request.getInputArtifacts();
+    if (inputArtifacts == null || inputArtifacts.isEmpty()) {
+      throw new IllegalArgumentException("At least one input artifact must be provided to bake");
+    }
+    List<Artifact> artifacts = new ArrayList<>();
+    Artifact artifact = inputArtifacts.get(0);
```

Comment:
I wrote the fontend to pass us a single `inputArtifact` instead of a list of `inputArtifacts`. I'm happy to fix that in the frontend but I think it would be more clear to use a single artifact.

Reply:
Update the KustomizeBakeManifestRequest in order to allow only one input Artifact